unit_tests/test_socket_connection.py

```diff
@@ -553,10 +551,11 @@ def test_raw_l2_oversized(self):
 
         uut = SocketConnection(host="lo", proto="raw-l2")
```

Comment:
We should switch to the new classes here too. Else we get misleading depreciation warnings while running the tests.

Reply:
Imho, `SocketConnection` still should be tested at least for some scenarios, even if it is deprecated. Until it will be removed, we still need to know, if it is working properly. I believe that deprecation warnings when running tests are not a problem.

Reply:
Sounds reasonable. I also found a way to ignore the depreciation warnings for `SocketConnection` for PyCharm and pytest, so that's not a problem anymore.